src/openrct2/network/Network.cpp

```diff
@@ -965,20 +964,31 @@ bool Network::CheckSRAND(uint32_t tick, uint32_t srand0)
 void Network::CheckDesynchronizaton()
 {
     // Check synchronisation
-    if (GetMode() == NETWORK_MODE_CLIENT && !_desynchronised && !CheckSRAND(gCurrentTicks, scenario_rand_state().s0))
+    if (GetMode() == NETWORK_MODE_CLIENT && !_desynchronised)
     {
-        _desynchronised = true;
+        if (CheckSRAND(gCurrentTicks, scenario_rand_state().s0) == false)
+        {
+            if (_desyncStarted == true)
+            {
+                _desynchronised = true;
 
-        char str_desync[256];
-        format_string(str_desync, 256, STR_MULTIPLAYER_DESYNC, nullptr);
+                char str_desync[256];
+                format_string(str_desync, 256, STR_MULTIPLAYER_DESYNC, nullptr);
 
-        auto intent = Intent(WC_NETWORK_STATUS);
-        intent.putExtra(INTENT_EXTRA_MESSAGE, std::string{ str_desync });
-        context_open_intent(&intent);
+                auto intent = Intent(WC_NETWORK_STATUS);
+                intent.putExtra(INTENT_EXTRA_MESSAGE, std::string{ str_desync });
+                context_open_intent(&intent);
 
-        if (!gConfigNetwork.stay_connected)
+                if (!gConfigNetwork.stay_connected)
+                {
+                    Close();
+                }
+            }
+            _desyncStarted = true;
```

Comment:
It'd be nice to explain in a comment why this bool is necessary, possible with link to this PR. For someone who doesn't know what this is about, it appears to be simply skipping one tick before truely desyncing.